src/script/components/message/MessageWrapper.tsx

```diff
@@ -0,0 +1,399 @@
+  isLastDeliveredMessage: boolean;
+  isMarked: boolean;
+  isSelfTemporaryGuest: boolean;
+  /** The last read timestamp at the moment the conversation was rendered */
+  lastReadTimestamp: number;
+  message: BaseMessage;
+  messageActions: {
+    deleteMessage: (conversation: Conversation, message: BaseMessage) => void;
+    deleteMessageEveryone: (conversation: Conversation, message: BaseMessage) => void;
+  };
+  messageRepository: MessageRepository;
+  onVisible?: () => void;
+  previousMessage?: BaseMessage;
+  selfId: QualifiedId;
+  shouldShowInvitePeople: boolean;
+  teamState?: TeamState;
+}
+
+function useRelativeTimestamp(timestamp: number, asDay?: boolean) {
+  const calculateTimestamp = (ts: number, isDay: boolean) => {
+    const date = fromUnixTime(ts / TIME_IN_MILLIS.SECOND);
+    if (isYoungerThan2Minutes(date)) {
+      return t('conversationJustNow');
+    }
+
+    if (isYoungerThan1Hour(date)) {
+      return fromNowLocale(date);
+    }
+
+    if (isToday(date)) {
+      const time = formatTimeShort(date);
+      return isDay ? `${t('conversationToday')} ${time}` : time;
+    }
+
+    if (isYesterday(date)) {
+      return `${t('conversationYesterday')} ${formatTimeShort(date)}`;
+    }
+    if (isYoungerThan7Days(date)) {
+      return formatLocale(date, 'EEEE p');
+    }
+
+    const weekDay = formatLocale(date, 'EEEE');
+    const dayMonth = formatDayMonth(date);
+    const year = isThisYear(date) ? '' : ` ${date.getFullYear()}`;
+    const time = formatTimeShort(date);
+    return isDay ? `${weekDay}, ${dayMonth}${year}, ${time}` : `${dayMonth}${year}, ${time}`;
+  };
+  const [timeago, setTimeago] = useState<string>(calculateTimestamp(timestamp, asDay));
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setTimeago(calculateTimestamp(timestamp, asDay));
+    }, TIME_IN_MILLIS.MINUTE);
+    return () => clearInterval(interval);
+  });
+  return timeago;
+}
+
+const MessageWrapper: React.FC<MessageParams & {hasMarker: boolean}> = ({
+  message,
+  conversation,
+  selfId,
+  hasMarker,
+  isSelfTemporaryGuest,
+  isLastDeliveredMessage,
+  shouldShowInvitePeople,
+  previousMessage,
+  hasReadReceiptsTurnedOn,
+  onClickAvatar,
+  onClickImage,
+  onClickInvitePeople,
+  onClickLikes,
+  onClickMessage,
+  onClickTimestamp,
+  onClickParticipants,
+  onClickReceipts,
+  onClickResetSession,
+  onClickCancelRequest,
+  onLike,
+  messageRepository,
+  messageActions,
+  teamState = container.resolve(TeamState),
+}) => {
+  const findMessage = (conversation: Conversation, messageId: string) => {
+    return messageRepository.getMessageInConversationById(conversation, messageId, true, true);
+  };
+  const clickButton = (message: CompositeMessage, buttonId: string) => {
+    if (message.selectedButtonId() !== buttonId && message.waitingButtonId() !== buttonId) {
+      message.waitingButtonId(buttonId);
+      messageRepository.sendButtonAction(conversation, message, buttonId);
+    }
+  };
```

Comment:
could these functions be moved to a utils file possibly? not sure since they are referencing the messageRepository if it makes sense. thinking maybe they dont have to be passed as params.